File: reversing-consonants-from-a-string.rb
def reverse_consonants(str) 
	str_array = str.split("")
	consonants = 'bcdfghjklnpqrstvwxyzBCDFGHJKLNPQRSTVWXYZ'
	left = 0;
	right = str_array.size - 1;

	while(left < right)
		while(left < right && !consonants.include?(str_array[left]))
			left += 1
		end	

		while(left < right && !consonants.include?(str_array[right])) 
			right -= 1
		end

		str_array[left], str_array[right] = str_array[right], str_array[left]	

		left += 1
		right -= 1
	end	
	return str_array.join('')

end
puts reverse_consonants('somestring')

File: reversing-consonants-from-a-string.js
// We are required to write a JavaScript function that takes in a string of lowercase english alphabets as the only argument.

// The function should construct a new string in which the order of consonants is reversed and the vowels hold their relative positions.

// For example, if the input to the function is −

const reverseConsonants = (s) => {
	const consonants = 'bcdfghjklnpqrstvwxyzBCDFGHJKLNPQRSTVWXYZ';
	let sArray = s.split('');
	let left = 0;
	let right = sArray.length -1;

	while(left < right) {
		while(left < right && !consonants.includes(sArray[left])) {
			left++;

		}
		while(left < right && !consonants.includes(sArray[right])) {
			right --;
		}
		[sArray[left], sArray[right]] = [sArray[right], sArray[left]];
		left++;
		right --;
	}

	return sArray.join('');
}

reverseConsonants('somestring')